Update on Multi-State Feedback to Make It Play Nicer with the Original Feedback

digitaldrummerj opened the pull request:
Based on the feedback from the community, it was suggested to redesign the multi-state feedback so that it played nicer with the original feedback system and did not take over the whole button when turned on.  This pull request is an attempt at the redesign and I would like feedback from the community on the design.  

At the bottom of this description, is the information on how you can test out these changes on your Companion v3 instance without having to install a new Companion version.

**Note:** If we want to get this into the Companion v3 final release, the deadline for that is July 5th to get it into the beta release.  Right now, if we do not do anything the existing beta with the preset advanced categories and the full button images will be what is released into Companion v3.

## Changes:

* Redesigned the icons to have both top and bottoms versions and not take up the whole button
* Integrated the new multi-state feedback into the participant presets as the last preset and removed the advanced preset button categories.  
* Removed from the preset participant buttons the original feedback for mic live, active speaker, and hand raised.  They are still available to manually add to a button.
* Added ability to either see all of the states on the button or just the active states
* To work out of the box for someone that is color blind, it was suggested to change the participant presets to have a black background and white text to allow us to have more contrast with the icons as well as to change the selected color be a gray instead of yellow so it does not blend with the hand raised icon.
* The multi-state feedback images will display a 72x72 version if the Companion topbar is turned off or a 72x58 version if the Companion topbar is turned on

## Sample of What The Buttons Look Like for the Bottom Icons

![image](https://github.com/bitfocus/companion-module-zoom-osc-iso/assets/708423/e12e4fe7-6bd2-4692-a199-392155e29862)

## Test Out The Changes Before Release

If you would like to test out the changes on your Companion v3 build then you can download the attached zip file.  Once you have the zip file then unzip this onto your Companion v3 computer, then click on the little cog in the upper right corner of the Companion GUI, then click the Select button for the Developer Modules path, and select the companion-module-test folder that is in the unzipped directory.  You should now be able to then test out the changes for the multi-state feedback on your machine without needing to install a new Companion release.

In the config for the module, you will see a new configuration option called "Participant Multi-State Feedback Image Design" with several options under it

1. **Icons at the Top (Only Active States)** - this will show the icons at the top of the button and only the states that are active.  This is the default if you have not selected an option before.  This is also what the presets are designed for
2. **Icons at the Top (On and Off States)** - this will show the icons at the top and show both the on and off states
3. **Icons at the Bottom (Only Active States)** - this will show the icons at the bottom of the button and only the states that are active
4. **Icons at the Bottom (On and Off States)** - this will show the icons at the bottom of the button and show both the on and off states
5. **Turn Off Multi-State feedback** - as there is no way in a preset to include a displayed feedback, this is the suggested option from the Companion team to be able to include the multi-state feedback in the presets but allow you to global turn them off so that they will not display the multi-state image and will allow the original hand raised feedback png show.

### Revert back to the built-in module after testing

If you want to revert back to the built-in version of the Companion module, click the little cog in the upper right again and it will restart Companion and use the built-in modules.

[Companion v3 ZoomOSC-ISO Multi-State Feedback Test Using ZoomTestKitchen Design.zip](https://github.com/bitfocus/companion-module-zoom-osc-iso/files/11877715/Companion.v3.ZoomOSC-ISO.Multi-State.Feedback.Test.Using.ZoomTestKitchen.Design.zip)

ZoomTestKitchen commented:
After much review and exchange of comments with James, I'm attaching my recommended layout.

A few previous comments included:

1. In my experience, a red microphone does not commonly indicate a LIVE mic, but rather a MUTED mic.
2. Separate the BLUE Active Speaker bar from the icons. If the icons are on top, put the blue bar on the bottom, and vice versa.
3. I prefer the blue active speaker bar on the top as it is easy to see and only requires a single line break to separate the text from overlapping.
4. A fixed font size creates a consistent button layout rather than a mix of font sizes driven by the amount of text on the button.
5. Top-Left alignment of the text provides as many characters from the beginning as possible.

These are the specifics of my submitted layout.

- Font Size = 7pt
- Single line break at the Top
- Dynamic text, top-left aligned.
- Unselected Background Color: Hex Code #000000
- Selected Background Color: Hex Code #484848
- Green Mic Color: Hex Code #30F83E
- Hand Skin Tone 1: Hex Code #F19F55
- Hand Skin Tone 2: Hex Code #AD6A2D
- Icon Max Height (Mic/Cam) = 18 pixels
- Icons Location: Bottom Row
- Active Speaker Indicator Location: Top Row
- Active Speaker Indicator Color: Hex Code #0003FF
- Active Speaker Bar Height: 5 pixels

Here is a description of the button images that I have submitted:

**Row 1**
1. Button 01 - Mic ON, Cam OFF, Hand NOT RAISED, Active Speaker NO, Selected NO
2. Button 02 - Mic ON, Cam OFF, Hand NOT RAISED, Active Speaker YES, Selected NO
3. Button 03 - Mic ON, Cam ON, Hand NOT RAISED, Active Speaker NO, Selected NO
4. Button 04 - Mic ON, Cam ON, Hand NOT RAISED, Active Speaker YES, Selected NO
5. Button 05 - Mic ON, Cam ON, Hand RAISED, Active Speaker NO, Selected NO
6. Button 06 - Mic ON, Cam ON, Hand RAISED, Active Speaker YES, Selected NO 

**Row 2**
1. Button 01 - Mic ON, Cam OFF, Hand NOT RAISED, Active Speaker NO, Selected YES
2. Button 02 - Mic ON, Cam OFF, Hand NOT RAISED, Active Speaker YES, Selected NO
3. Button 03 - Mic ON, Cam ON, Hand NOT RAISED, Active Speaker NO, Selected NO
4. Button 04 - Mic ON, Cam ON, Hand NOT RAISED, Active Speaker YES, Selected YES
5. Button 05 - Mic ON, Cam ON, Hand RAISED, Active Speaker NO, Selected NO
6. Button 06 - Mic ON, Cam ON, Hand RAISED, Active Speaker YES, Selected YES 

![Screenshot 2023-06-26 at 8 22 55 PM](https://github.com/bitfocus/companion-module-zoom-osc-iso/assets/124208331/17daa1d0-4a1d-4e33-b6d9-4d8ca8a69ac3)

digitaldrummerj commented:
@ZoomTestKitchen thank you for the feedback.  what is the hex code for your gray icons?

MuddySchlegel commented:
Thanks for taking the time to make the multi-feedback options available and more functional. I like the direction this is going. As a colorblind person who runs events with ZoomISO/ZoomOSC, I wanted to share some quick feedback. The color scheme @ZoomTestKitchen is proposing is generally fairly easy for me to differentiate.  I like the thin blue stripe at the top to indicate active speaker. 

I'm not sure if I prefer top- or bottom-aligned icons, but I don't love when the icons overlap with the text. I know it can be tough to avoid, but I wonder if the raised hand icon was scaled down to be the same vertical height as the mic and camera if that would help.

I prefer a uniformly-sized font for the buttons, so 7pt seems good to me.

I also wonder if the not-raised hand needs to be a visible icon or whether that would be better as no icon when hand not raised. I like the uniformity of all the buttons having the hand, but to me it seems to make the raised hand less noticeable than if the non-raised hands were just a blank spot.  Not sure, just thinking out loud.

Finally, I'm not sure if the "on" color of the mic, camera and raised hand is the same across all three icons in Jeff's graphic. They look the same to my colorblind eyes as some sort of green/yellow color. I like them all the same color, indicating a consistent "on" state color. and then gray for the "off" state. It's quite possible that those icons are not all the same color in Jeff's graphic which is why I mention it.

Anyway, thanks again for your work on this issue! Much appreciated.

Cheers, Muddy

digitaldrummerj commented:
@ZoomTestKitchen Here is a version of the feedback using your design above.  It has the presets design for the icons to be on the bottom and the default set of icons is to show all states.  I removed from the preset button the original feedback for mic live, active speaker, and hand raised (they are all still available to manually add to a button).  

I also updated the pull request code to this design.

[Companion v3 ZoomOSC-ISO Multi-State Feedback Test Using ZoomTestKitchen Design.zip](https://github.com/bitfocus/companion-module-zoom-osc-iso/files/11877667/Companion.v3.ZoomOSC-ISO.Multi-State.Feedback.Test.Using.ZoomTestKitchen.Design.zip)

digitaldrummerj commented:
for the font size, I tried out the new preset having 7 for the font size on an event today.  The uniformity was nice but I found the readability difficult with the font at 7.  I would prefer to have the preset as auto for the font size.

Here is a sample with the top row at 7 and the bottom row at auto.  For the auto font, I did need to remove the starting line break on the participant name.

![Participant at font size 7 vs 14](https://github.com/bitfocus/companion-module-zoom-osc-iso/assets/708423/c43d3e2a-2ab3-45e5-adfd-5af6e3d3cf30)

digitaldrummerj commented:
After a few hours of running the event with the participant name font at 7, I can live with having the preset font size for participants at 7 and I can change it if I need to in my profile.  It did take me longer to find the person I was looking for but it was not enough of a difference to push for a font size of auto vs 7 in the preset.

AndrewCarluccio commented:
We're going to finish testing this out and give some feedback shortly

digitaldrummerj commented:
I have 2 events this week with the new design and both times having the non-hand raised icon (the gray one) has made me think that someone has their hand up when the gray icon was showing.  I am thinking it is due to the size that I am thinking that someone has their hand raised when they don't so I did a design with a smaller hand icon. 

![ZoomOSC Feedback Smaller Hand Raised](https://github.com/bitfocus/companion-module-zoom-osc-iso/assets/708423/21bd033c-ec6a-45d4-803e-1e1cb81276db)


Here is a pkg with the smaller hand-raised icon if you want to try it out using the instructions above in the description.

[Companion v3 ZoomOSC-ISO Multi-State Feedback Smaller Hand Raised Icon Pkg](https://github.com/bitfocus/companion-module-zoom-osc-iso/files/11901924/Companion.v3.ZoomOSC-ISO.Multi-State.Feedback.Smaller.Hand.Raised.Icon.zip)

I have 2 more events this week that I will try it out on and see how it works/looks but I wanted to put this up here so other could provide feedback if they would like.

ZoomTestKitchen commented:
Thanks for this feedback. I can understand that. If you find that the
smaller hand doesn’t do the trick, I would recommend removing the hand, and
just have it turn-on and turn-off.

Although this by itself would look a little strange as far as visual
balance is concerned. To help with this, I would recommend moving the
camera to the right side, leave the mic on the left side, and have the hand
in the middle. Then it will always look balanced.



On Wed, Jun 28, 2023 at 11:28 PM Justin James ***@***.***>
wrote:

> I have 2 events this week with the new design and both times having the
> non-hand raised icon (the gray one) has made me think that someone has
> their hand up when the gray icon was showing. I am thinking it is due to
> the size that I am thinking that someone has their hand raised when they
> don't so I did a design with a smaller hand icon.
>
> [image: ZoomOSC Feedback Smaller Hand Raised]
> <https://user-images.githubusercontent.com/708423/249671138-21bd033c-ec6a-45d4-803e-1e1cb81276db.png>
>
> Here is a pkg with the smaller hand-raised icon if you want to try it out
> using the instructions above in the description.
>
> Companion v3 ZoomOSC-ISO Multi-State Feedback Smaller Hand Raised Icon Pkg
> <https://github.com/bitfocus/companion-module-zoom-osc-iso/files/11901924/Companion.v3.ZoomOSC-ISO.Multi-State.Feedback.Smaller.Hand.Raised.Icon.zip>
>
> I have 2 more events this week that I will try it out on and see how it
> works/looks but I wanted to put this up here so other could provide
> feedback if they would like.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bitfocus/companion-module-zoom-osc-iso/pull/126#issuecomment-1612413206>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A5TUJS6PTX45HUFTRCM5P43XNT75LANCNFSM6AAAAAAZUX46SY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 


*Jeff Widgren*
*email • ***@***.***
*youtube • *Zoom Test Kitchen YouTube Channel
<https://www.youtube.com/c/ZoomTestKitchen>
*mobile • *(913) 908-8485

*Zoom Solution Consultant & Developer*
*Improving Management & Automation for Zoom Meetings/Webinars*

digitaldrummerj commented:
@MuddySchlegel Thank you for the feedback during our chat today about the hand-raised icon off and on state looking the same to you being color blind and asking if we could add an option where the mic and camera are always on and the hand raised only shows when it is raised.  I have added this as an option that you can select in the config settings in addition to the existing bottom layouts of only on states and all states.

![image](https://github.com/bitfocus/companion-module-zoom-osc-iso/assets/708423/6b97dafc-4077-4a7f-8ae9-fe9a92f3e895)

Here is an updated package that you can test out

[Companion v3 ZoomOSC-ISO Multi-State Feedback - Hand Raised Only Show When Active.zip](https://github.com/bitfocus/companion-module-zoom-osc-iso/files/11911021/Companion.v3.ZoomOSC-ISO.Multi-State.Feedback.-.Hand.Raised.Only.Show.When.Active.zip)

@ZoomTestKitchen and @Muddy - As part of this update, I also update the multi-state feedback to be more descriptive instead   of just saying advanced as that was confusing for new users as to what advanced feedback meant.

![image](https://github.com/bitfocus/companion-module-zoom-osc-iso/assets/708423/f3a5b166-9ce6-4258-aa2a-ec4846ceb953)

MuddySchlegel commented:
Thank you, @digitaldrummerj! After our chat I got right into a client event, so I greatly appreciate you capturing my feedback for this thread! I look forward to testing the latest version with my colorblindness!  Cheers, Muddy

digitaldrummerj commented:
@AndrewCarluccio checking in on if you have been able to test it out and if you had any feedback?  I am also happy to hop on a call and do a quick demo if that would be quicker.

digitaldrummerj commented:
@AndrewCarluccio  you can view all of the images here.  I think they are all right but would be good to have a 2nd set of eyes on them to make sure they all look good since the code has them as base64 images.


* 72x58 - https://www.canva.com/design/DAFm3g3c6KA/rEYRYtqcvp3FewXnxNvMpw/edit?utm_content=DAFm3g3c6KA&utm_campaign=designshare&utm_medium=link2&utm_source=sharebutton
* 72x72 - https://www.canva.com/design/DAFm3oEawFY/moBny6ZFPfalVAoTt598bQ/edit?utm_content=DAFm3oEawFY&utm_campaign=designshare&utm_medium=link2&utm_source=sharebutton

There are 3 sets of images: 

1. only show active states (mic on, camera on, hand raised)
2. show all states for both on and off for mic, camera and hand raised
3. show mic/camera on and off and only hand when raised

AndrewCarluccio reviewed (approved):
Thanks for the PR. Zoom team checked it out, and looks good from our side.

digitaldrummerj commented:
@JeffreyDavidsz can you merge this since @AndrewCarluccio has approved it?
